packages/connect/src/connector/index.ts

```diff
@@ -57,9 +34,65 @@ export interface Config {
  * extension or by executing a light client directly from JavaScript, depending on whether the
  * extension is installed and available.
  */
-export function createScClient(config?: Config) {
-  const forceEmbedded = config?.forceEmbeddedNode
+export const createScClient = (config?: Config): ScClient => {
+  if (config?.forceEmbeddedNode || typeof document !== "object")
+    return smoldotScClient(config?.embeddedNodeConfig)
+
+  const client = isExtensionPresent().then((isPresent) =>
+    isPresent
+      ? extensionScClient()
+      : smoldotScClient(config?.embeddedNodeConfig),
+  )
+
+  return {
+    async addChain(chainSpec, jsonRpcCallback, databaseContent) {
+      return (await client).addChain(
+        chainSpec,
+        jsonRpcCallback,
+        databaseContent,
+      )
+    },
+    async addWellKnownChain(id, jsonRpcCallback, databaseContent) {
+      return (await client).addWellKnownChain(
+        id,
+        jsonRpcCallback,
+        databaseContent,
+      )
+    },
+  }
+}
+
+async function isExtensionPresent() {
+  let isPresent = false
+  const onMessage = ({ source, origin, data }: MessageEvent) => {
+    if (
+      source !== window ||
+      origin !== window.origin ||
+      typeof data !== "object" ||
+      data.origin !== "substrate-connect-extension" ||
+      data.type !== "is-extension-present"
+    )
+      return
+    isPresent = true
+    window.removeEventListener("message", onMessage)
+  }
+  window.addEventListener("message", onMessage)
+  window.postMessage({
+    origin: "substrate-connect-client",
+    type: "is-extension-present",
+  })
```

Comment:
@josepot Do you think this message should be added to the `@substrate/connect-extension-protocol`?